nixos/modules/services/networking/syncthing.nix

```diff
@@ -141,7 +141,14 @@ let
        ${jq} -e .requiresRestart > /dev/null; then
         curl -X POST ${curlAddressArgs "/rest/system/restart"}
     fi
-  '');
+  ''
+  +
+  (lib.optionalString (cfg.guiPasswordFile != null) ''
+     pw_bcrypt=$(${pkgs.mkpasswd}/bin/mkpasswd -m bcrypt --stdin <"${cfg.guiPasswordFile}") \
+     curl -X PATCH --variable '%pw_bcrypt' --expand-json '{ "password": "{{pw_bcrypt}}" }' \
```

Comment:
Although not clear-text, it would be nice if `$pw_bcrypt` wasn't visible in the process listings. Can't the arguments be put in a file (prepared with `replace-secret`) and then run with `curl -K ./secret_params.txt ...`?

Reply:
Thanks, that's a good point. -K i couldn't get working, I think in part because the curl here is actually a bash function that wraps real curl, but now that I think about it, I didn't try with just the one json argument in that file. I found curl allows variables from file though, which works for me and I think this resolves the issue about visibility in process listings.